src/babel-plugin-bare-specifiers.ts

```diff
@@ -76,19 +77,22 @@ export const resolveBareSpecifiers = (
       // If we have an import that crosses the top-level / componentDir
       // boundary, fix up the path to generate polyserve-style ../ cross-package
       // imports.
-      if (componentRequestRelationship !== undefined &&
+      if (isComponentRequest !== undefined &&
           componentDir !== undefined && rootDir !== undefined &&
           pathIsInside(resolvedSpecifier, componentDir) &&
           !pathIsInside(filePath, componentDir)) {
+        if (packageName === undefined) {
+          throw new Error(
+              'resolveBareSpecifiers(): packageName must be set when isComponentRequest === true');
+        }
         const relativePathFromComponentDir =
             relative(componentDir, resolvedSpecifier);
-        // Important: Create a URL pointing one level up from the root dir
+        // Important: Create a URL pointingup from the root dir
+        const relativeDistance =
```

Comment:
This is ok for for now, but I was hoping to get to get away from detecting the `@` in the package name and do the operations all based on paths. I think a refactoring that pulls out all off these settings into some kind of callback to customize would be nice.

Reply:
gotcha, I didn't understand that. This is all new logic to me so I'm less familiar with it, if you can give some guidance on how that would work I can add it now.

Reply:
Don't worry about it for now, I think it might require a little more logic and this works, so let's lock that in. I'm not sure the value would be there now.